tree/ntuple/v7/inc/ROOT/RColumnElementBase.hxx

```diff
@@ -0,0 +1,171 @@
+   virtual void Pack(void *destination, const void *source, std::size_t count) const
+   {
+      std::memcpy(destination, source, count);
+   }
+
+   /// If the on-storage layout and the in-memory layout differ, unpacking creates a memory page from an on-storage page
+   virtual void Unpack(void *destination, const void *source, std::size_t count) const
+   {
+      std::memcpy(destination, source, count);
+   }
+
+   std::size_t GetSize() const { return fSize; }
+   std::size_t GetBitsOnStorage() const { return fBitsOnStorage; }
+   std::size_t GetPackedSize(std::size_t nElements = 1U) const { return (nElements * fBitsOnStorage + 7) / 8; }
+}; // class RColumnElementBase
+
+// All supported C++ in-memory types
+enum class EColumnCppType {
+   kChar,
+   kBool,
+   kByte,
+   kUint8,
+   kUint16,
+   kUint32,
+   kUint64,
+   kInt8,
+   kInt16,
+   kInt32,
+   kInt64,
+   kFloat,
+   kDouble32,
+   kDouble,
+   kClusterSize,
+   kColumnSwitch,
+};
+
+std::unique_ptr<RColumnElementBase> GenerateColumnElement(EColumnCppType cppType, EColumnType colType);
+
+template <typename CppT>
+std::unique_ptr<RColumnElementBase> RColumnElementBase::Generate(EColumnType type)
+{
+   if constexpr (std::is_same_v<CppT, char>)
+      return GenerateColumnElement(EColumnCppType::kChar, type);
+   else if constexpr (std::is_same_v<CppT, bool>)
+      return GenerateColumnElement(EColumnCppType::kBool, type);
+   else if constexpr (std::is_same_v<CppT, std::byte>)
+      return GenerateColumnElement(EColumnCppType::kByte, type);
+   else if constexpr (std::is_same_v<CppT, std::uint8_t>)
+      return GenerateColumnElement(EColumnCppType::kUint8, type);
+   else if constexpr (std::is_same_v<CppT, std::uint16_t>)
+      return GenerateColumnElement(EColumnCppType::kUint16, type);
+   else if constexpr (std::is_same_v<CppT, std::uint32_t>)
+      return GenerateColumnElement(EColumnCppType::kUint32, type);
+   else if constexpr (std::is_same_v<CppT, std::uint64_t>)
+      return GenerateColumnElement(EColumnCppType::kUint64, type);
+   else if constexpr (std::is_same_v<CppT, std::int8_t>)
+      return GenerateColumnElement(EColumnCppType::kInt8, type);
+   else if constexpr (std::is_same_v<CppT, std::int16_t>)
+      return GenerateColumnElement(EColumnCppType::kInt16, type);
+   else if constexpr (std::is_same_v<CppT, std::int32_t>)
+      return GenerateColumnElement(EColumnCppType::kInt32, type);
+   else if constexpr (std::is_same_v<CppT, std::int64_t>)
+      return GenerateColumnElement(EColumnCppType::kInt64, type);
+   else if constexpr (std::is_same_v<CppT, float>)
+      return GenerateColumnElement(EColumnCppType::kFloat, type);
+   else if constexpr (std::is_same_v<CppT, double>)
+      return GenerateColumnElement(EColumnCppType::kDouble, type);
+   else if constexpr (std::is_same_v<CppT, Double32_t>)
+      return GenerateColumnElement(EColumnCppType::kDouble32, type);
```

Comment:
`Double32_t` is a `typedef`, so for what concerns C++ it is a `double` and this check should never be true.